implementations/apollo-server/index.js

```diff
@@ -57,6 +57,26 @@ const resolvers = {
   User: {
     name() {
       return "Jane Smith";
+    },
+    /** @type {(user: { email: String, totalProductsCreated: Number, yearsOfEmployment: Number }, args: any, context: any) => any} */
+    averageProductsCreatedPerYear: (user, args, context) => {
+      if (user.email != "support@apollographql.com") {
+        throw new ApolloError("user.email was not 'support@apollographql.com'")
+      }
+      return Math.round(user.totalProductsCreated / user.yearsOfEmployment);
+    },
+    /** @type {(reference: any) => any} */
+    __resolveReference: (reference) => {
+      if (reference.email) {
+        const user = { email: reference.email, name: "Jane Smith", totalProductsCreated: 1337 };
+        if (reference.yearsOfEmployment) {
```

Comment:
shouldn't this field be marked as required as well so that way the router sends this field?